### PR TITLE
prost-build: Extract file descriptor loading from compile_protos()

### PR DESCRIPTION
**Pull request by swallez**
Extracts the file descriptor loading part from `prost_build::compile_protos()` to a new `prost_build::load_fds()` function.

This allows inspection or even modification of descriptors before calling `prost_build::compile_fds`. Possible use cases include generating additional files from the descriptors, or set computed `Config` attributes by introspecting what is actually present in the `.proto` files before generating the code.

### CODE REVIEW

File: prost-build/src/config.rs [hunk 1]
```diff
@@ -835,37 +835,31 @@ impl Config {
...
-    ///   prost_build.btree_map(&["."]);
-    ///   prost_build.compile_protos(&["src/frontend.proto", "src/backend.proto"], &["src"])?;
-    ///   Ok(())
+    /// # use prost_types::FileDescriptorSet;
+    /// # use prost_build::Config;
+    /// fn main() -> std::io::Result<()> {
+    ///   let mut config = Config::new();
+    ///   let file_descriptor_set = config.load_fds(&["src/frontend.proto", "src/backend.proto"], &["src"])?;
+    ///
+    ///   // Inspect file_descriptor_set and tweak config
```
Comment (caspermeijn) [inline prost-build/src/config.rs:852]: Can you change this to a concrete example?

File: prost-build/src/config.rs [hunk 2]
```diff
@@ -835,37 +835,31 @@ impl Config {
...
-    ///
-    /// This method is like the `prost_build::compile_protos` function, with the added ability to
-    /// specify non-default code generation options. See that function for more information about
-    /// the arguments and generated outputs.
-    ///
-    /// The `protos` and `includes` arguments are ignored if `skip_protoc_run` is specified.
+    /// Loads `.proto` files as a `FileDescriptorSet`. This allows inspection of the descriptors
+    /// before calling `prost_build::compile_fds`. A possible use case it to compute `Config`
+    /// attributes by introspecting what is actually present in the `.proto` files before
+    /// generating the code.
```
Comment (caspermeijn) [inline prost-build/src/config.rs:841]: I suggest rewording this a bit. The square brackets cause `cargo doc` to create a link to the type.

```suggestion
    /// Loads `.proto` files as a [`FileDescriptorSet`]. This allows inspection of the descriptors
    /// before calling [`Config::compile_fds`]. This could be used to change [`Config`] 
    /// attributes after introspecting what is actually present in the `.proto` files.
```

File: prost-build/src/config.rs (hunk 1 above)
Comment (swallez) [reply prost-build/src/config.rs]: Done in cbf0813.

**Comment by swallez** [comment]
@caspermeijn Thanks for the review. I've added a usage example in the `test_generate_message_attributes` test and to the docs.

**Review by caspermeijn — APPROVED** [review]
Thank you for your contribution

**Comment by swallez** [comment]
Thanks @caspermeijn! What is the next step for this PR to be merged?

**Comment by caspermeijn** [comment]
> Thanks @caspermeijn! What is the next step for this PR to be merged?

There you go
